core-web/libs/sdk/experiments/src/lib/hooks/DotExperimentsProvider.tsx

```diff
@@ -0,0 +1,78 @@
+import { ReactElement, ReactNode, useEffect, useState } from 'react';
+
+import { isInsideEditor } from '@dotcms/client';
+
+import DotExperimentsContext from './DotExperimentsContext';
+
+import { DotExperiments } from '../dot-experiments';
+import { DotExperimentConfig } from '../shared/models';
+
+interface DotExperimentsProviderProps {
+    children?: ReactNode;
+    config: DotExperimentConfig;
+}
+
+/**
+ * `DotExperimentsProvider` is a component that uses React's Context API to provide
+ * an instance of `DotExperiments` to all of its descendants.
+ *
+ *
+ * @component
+ * @example
+ * ```jsx
+ *
+ * // Your application component
+ * function App() {
+ *
+ * // Other configuration options will be taken from environment variables.
+ *  const config = {
+ *     'api-key': process.env.NEXT_PUBLIC_EXPERIMENTS_API_KEY ,
+ *     server: process.env.NEXT_PUBLIC_DOTCMS_HOST ,
+ *     debug: process.env.NEXT_PUBLIC_EXPERIMENTS_DEBUG,
+ *     redirectFn: YourRedirectFunction
+ *   };
+
+ *
+ *   return (
+ *     <DotExperimentsProvider config={config}>
+ *       <YourDescendantComponent />
```

Comment:
Should we have here the DotLayout component?

Reply:
Yes, because we are using @dotcms/client to get data and all necessary to run Dotcms as SPA, I think.

Added.